src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -1508,9 +1501,13 @@ namespace QgsWfs
           QDomElement bbElem = doc.createElement( QStringLiteral( "gml:boundedBy" ) );
           QDomElement boxElem = QgsOgcUtils::rectangleToGMLEnvelope( &box, doc, params.srsName, params.hasAxisInverted, prec );
 
-          if ( crs.isValid() )
+          if ( crs.isValid() && params.srsName.isEmpty() )
+          {
+            boxElem.setAttribute( QStringLiteral( "srsName" ), crs.authid() );
+            gmlElem.setAttribute( QStringLiteral( "srsName" ), crs.authid() );
+          }
+          else if ( !params.srsName.isEmpty() )
```

Comment:
why removing "boxElem.setAttribute( QStringLiteral( "srsName" ), params.srsName );" when !params.srsName.isEmpty() ?

Reply:
Because it is already done in `QgsOgcUtils::rectangleToGMLEnvelope`, the `srsName` attribute is already set to `params.srsName` if not empty